### PR TITLE
Feature/feature google calendar

### PR DESCRIPTION
**Pull request by marclll**

### CODE REVIEW

File: src/test/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+            + "  \"id\": \"idtest\",\n"
+            + "  \"summary\":\"Event for tests\",\n"
+            + "  \"location\": \"Wherever you wanna be\",\n"
+            + "  \"description\": \"This is an event I created just to do tests\",\n"
+            + "  \"color\": \"11\",\n"
+            + "  \"emailReminderMinutes\": 10,\n"
+            + "  \"repetitionInterval\": 1,\n"
+            + "  \"startDate\": \"2020-04-26T18:20:38\",\n"
+            + "  \"endDate\": \"2020-04-26T23:20:38\"\n"
+            + "}";
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestControllerTest.java:71]: Codacy Issue found: [The String "}" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 17
           

Complexity increasing per file
==============================
- src/main/java/org/pesmypetcare/webservice/entity/EventEntity.java  1
- src/main/java/org/pesmypetcare/webservice/jsonhandler/DateTimeHandler.java  2
- src/test/java/org/pesmypetcare/webservice/service/GoogleCalendarServiceTest.java  1
- src/main/java/org/pesmypetcare/webservice/error/CalendarAccessException.java  1
- src/test/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestControllerTest.java  1
- src/main/java/org/pesmypetcare/webservice/error/InitializationException.java  1
- src/main/java/org/pesmypetcare/webservice/service/GoogleCalendarServiceImpl.java  1
- src/main/java/org/pesmypetcare/webservice/dao/GoogleCalendarDaoImpl.java  4
- src/main/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestController.java  1
- src/main/java/org/pesmypetcare/webservice/factories/FirebaseFactory.java  2
- src/main/java/org/pesmypetcare/webservice/factories/CalendarServiceFactory.java  2
- src/test/java/org/pesmypetcare/webservice/dao/GoogleCalendarDaoTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404&bid=17713034)

File: src/main/java/org/pesmypetcare/webservice/entity/EventEntity.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        this.location = location;
+        this.description = description;
+        this.color = color;
+        this.emailReminderMinutes = emailReminderMinutes;
+        this.repetitionInterval = repetitionInterval;
+        this.startDate = startDate;
+        this.endDate = endDate;
+    }
+
+    public Event convertToEvent() {
```
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/entity/EventEntity.java:56]: Codacy Issue found: [Method length is 21 lines (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/test/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    private static String petName;
+    private static String urlBase;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private GoogleCalendarService service;
+
+    @BeforeAll
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestControllerTest.java:53]: Codacy Issue found: [Method length is 22 lines (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/main/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestController.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+     * @param petName Name of the pet the calendar belongs to
+     * @param body Body of the request containing the id of the event to retrieve with key eventId assigned
+     * @return Event retrieved
+     * @throws CalendarAccessException If an error occurs when accessing the calendar
+     * @throws DatabaseAccessException If an error occurs when accessing the database
+     */
+    @GetMapping("/event/{owner}/{petName}")
+    public EventEntity retrieveEvent(@RequestHeader(TOKEN) String accessToken, @PathVariable String owner,
+                              @PathVariable String petName, @RequestBody Map<String, Object> body)
+        throws CalendarAccessException, DatabaseAccessException{
```
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestController.java:106]: Codacy Issue found: ['{' is not preceded with whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/test/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+    public static void setUp() {
+        owner = "My owner";
+        petName = "My petName";
+        accessToken = "tokem";
+        eventList = new ArrayList<>();
+        eventEntity = new EventEntity();
+        urlBase = "/calendar";
+        jsonEventEntity = "{\n"
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestControllerTest.java:61]: Codacy Issue found: [The String "{\n" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/test/java/org/pesmypetcare/webservice/service/GoogleCalendarServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+        eventList = new ArrayList<>();
+        eventEntity = new EventEntity();
+        eventEntity.setId("My eventId");
+        eventEntity.setStartDate(new DateTime(date));
+        eventEntity.setEndDate(new DateTime(date2));
+        event = new Event();
+        event.setId("My eventId");
+        EventDateTime start = new EventDateTime()
+            .setDateTime(new DateTime(date))
+            .setTimeZone("Europe/Madrid");
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/service/GoogleCalendarServiceTest.java:72]: Codacy Issue found: [The String "Europe/Madrid" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/test/java/org/pesmypetcare/webservice/service/GoogleCalendarServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+    @InjectMocks
+    private GoogleCalendarService service = new GoogleCalendarServiceImpl();
+
+    @BeforeAll
+    public static void setUp() {
+        owner = "My owner";
+        petName = "My petName";
+        date = "2020-02-13T10:30:00";
+        date2 = "2020-02-13T12:30:00";
+        eventId = "My eventId";
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/service/GoogleCalendarServiceTest.java:60]: Codacy Issue found: [The String "My eventId" appears 3 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/main/java/org/pesmypetcare/webservice/entity/EventEntity.java [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+        Event event = new Event()
+            .setId(id)
+            .setSummary(summary)
+            .setLocation(location)
+            .setDescription(description)
+            .setColorId(color);
+
+        EventDateTime start = new EventDateTime()
+            .setDateTime(startDate)
+            .setTimeZone("Europe/Madrid");
```
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/entity/EventEntity.java:66]: Codacy Issue found: [The String "Europe/Madrid" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/main/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestController.java [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+        return googleCalendarService.retrieveEvent(accessToken, owner, petName, (String) body.get("eventId"));
```
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestController.java:107]: Codacy Issue found: [The String "eventId" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/main/java/org/pesmypetcare/webservice/entity/EventEntity.java [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+        id = event.getId();
+        summary = event.getSummary();
+        location = event.getLocation();
+        description = event.getDescription();
+        color = event.getColorId();
+        startDate = event.getStart().getDateTime();
+        endDate = event.getEnd().getDateTime();
+    }
+
+    public EventEntity(String id, String summary, String description, String location, String color,
```
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/entity/EventEntity.java:42]: Codacy Issue found: [More than 5 parameters (found 9).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/test/java/org/pesmypetcare/webservice/service/GoogleCalendarServiceTest.java [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+
+    @Mock
+    private GoogleCalendarDao googleCalendarDao;
+    @Mock
+    private PetDao petDao;
+
+    @InjectMocks
+    private GoogleCalendarService service = new GoogleCalendarServiceImpl();
+
+    @BeforeAll
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/service/GoogleCalendarServiceTest.java:54]: Codacy Issue found: [Method length is 24 lines (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/test/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestControllerTest.java [hunk 12]
```diff
@@ -0,0 +1,140 @@
...
+            + "}";
+        jsonEventId = "{\n"
+            + "  \"eventId\": \"idtest\"\n"
+            + "}";
+    }
+
+    @Test
+    public void createSecondaryCalendarShouldReturnStatusOk() throws Exception {
+        willDoNothing().given(service).createSecondaryCalendar(anyString(), anyString(), anyString());
+        mockMvc.perform(post(urlBase + "/" + owner + "/" + petName)
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestControllerTest.java:80]: Codacy Issue found: [The String "/" appears 10 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/test/java/org/pesmypetcare/webservice/service/GoogleCalendarServiceTest.java (hunk 11 above)
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/service/GoogleCalendarServiceTest.java:54]: Codacy Issue found: [NCSS for this method is 19 (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/test/java/org/pesmypetcare/webservice/dao/GoogleCalendarDaoTest.java [hunk 13]
```diff
@@ -0,0 +1,171 @@
...
+    @Mock
+    private com.google.api.services.calendar.Calendar.Events.Insert insertEvent;
+
+    @InjectMocks
+    private GoogleCalendarDao googleCalendarDao = new GoogleCalendarDaoImpl();
+
+    @BeforeAll
+    public static void setUp() {
+        accessToken = "tokenparatests";
+        calendarId = "calendarId";
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/dao/GoogleCalendarDaoTest.java:69]: Codacy Issue found: [The String "calendarId" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 17
           

Complexity increasing per file
==============================
- src/main/java/org/pesmypetcare/webservice/jsonhandler/DateTimeHandler.java  2
- src/test/java/org/pesmypetcare/webservice/dao/GoogleCalendarDaoTest.java  1
- src/test/java/org/pesmypetcare/webservice/service/GoogleCalendarServiceTest.java  1
- src/main/java/org/pesmypetcare/webservice/error/CalendarAccessException.java  1
- src/main/java/org/pesmypetcare/webservice/error/InitializationException.java  1
- src/main/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestController.java  1
- src/main/java/org/pesmypetcare/webservice/entity/EventEntity.java  1
- src/main/java/org/pesmypetcare/webservice/firebaseservice/CalendarServiceFactory.java  2
- src/main/java/org/pesmypetcare/webservice/dao/GoogleCalendarDaoImpl.java  4
- src/test/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestControllerTest.java  1
- src/main/java/org/pesmypetcare/webservice/service/GoogleCalendarServiceImpl.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404&bid=17713034)

File: src/test/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestControllerTest.java (hunk 5 above)
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestControllerTest.java:61]: Codacy Issue found: [The String "{\n" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/main/java/org/pesmypetcare/webservice/entity/EventEntity.java (hunk 8 above)
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/entity/EventEntity.java:66]: Codacy Issue found: [The String "Europe/Madrid" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/test/java/org/pesmypetcare/webservice/dao/GoogleCalendarDaoTest.java [hunk 14]
```diff
@@ -0,0 +1,171 @@
...
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/dao/GoogleCalendarDaoTest.java:69]: Codacy Issue found: [The String "calendarId" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/test/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestControllerTest.java (hunk 12 above)
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestControllerTest.java:80]: Codacy Issue found: [The String "/" appears 10 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/main/java/org/pesmypetcare/webservice/entity/EventEntity.java [hunk 15]
```diff
@@ -0,0 +1,87 @@
...
+    private String description;
+    private String color;
+    private Integer emailReminderMinutes;
+    private Integer repetitionInterval;
+    @JsonDeserialize(using = DateTimeHandler.class)
+    private DateTime startDate;
+    @JsonDeserialize(using = DateTimeHandler.class)
+    private DateTime endDate;
+
+    public EventEntity() {}
```
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/entity/EventEntity.java:30]: Codacy Issue found: ['}' is not preceded with whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/test/java/org/pesmypetcare/webservice/service/GoogleCalendarServiceTest.java (hunk 6 above)
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/service/GoogleCalendarServiceTest.java:72]: Codacy Issue found: [The String "Europe/Madrid" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/main/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestController.java (hunk 4 above)
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestController.java:106]: Codacy Issue found: ['{' is not preceded with whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/test/java/org/pesmypetcare/webservice/service/GoogleCalendarServiceTest.java (hunk 7 above)
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/service/GoogleCalendarServiceTest.java:60]: Codacy Issue found: [The String "My eventId" appears 3 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/test/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestControllerTest.java (hunk 3 above)
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestControllerTest.java:53]: Codacy Issue found: [Method length is 22 lines (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/test/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestControllerTest.java [hunk 16]
```diff
@@ -0,0 +1,140 @@
...
+        willReturn(eventList).given(service).getAllEventsFromCalendar(anyString(), anyString(), anyString());
+        mockMvc.perform(get(urlBase + "/" + owner + "/" + petName)
+            .header(TOKEN, accessToken))
+            .andExpect(status().isOk());
+    }
+
+    @Test
+    public void createEventShouldReturnStatusOk() throws Exception {
+        willDoNothing().given(service).createEvent(anyString(), anyString(), anyString(), isA(EventEntity.class));
+        mockMvc.perform(post(urlBase + "/event/" + owner + "/" + petName)
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestControllerTest.java:104]: Codacy Issue found: [The String "/event/" appears 4 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/main/java/org/pesmypetcare/webservice/dao/GoogleCalendarDaoImpl.java [hunk 17]
```diff
@@ -0,0 +1,112 @@
...
+    public void deleteSecondaryCalendar(String accessToken, String calendarId) throws CalendarAccessException {
+        service = factory.initializeService(accessToken);
+        try {
+            service.calendars().delete(calendarId).execute();
+        } catch (IOException e) {
+            throw new CalendarAccessException("Error deleting secondary calendar", e.getMessage());
+        }
+    }
+
+    @Override
```
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/dao/GoogleCalendarDaoImpl.java:49]: Codacy Issue found: [Method length is 16 lines (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/main/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestController.java (hunk 9 above)
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestController.java:107]: Codacy Issue found: [The String "eventId" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/test/java/org/pesmypetcare/webservice/service/GoogleCalendarServiceTest.java (hunk 11 above)
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/service/GoogleCalendarServiceTest.java:54]: Codacy Issue found: [NCSS for this method is 19 (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/test/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestControllerTest.java (hunk 1 above)
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestControllerTest.java:71]: Codacy Issue found: [The String "}" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/test/java/org/pesmypetcare/webservice/service/GoogleCalendarServiceTest.java (hunk 11 above)
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/service/GoogleCalendarServiceTest.java:54]: Codacy Issue found: [Method length is 24 lines (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/main/java/org/pesmypetcare/webservice/entity/EventEntity.java (hunk 2 above)
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/entity/EventEntity.java:56]: Codacy Issue found: [Method length is 21 lines (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/main/java/org/pesmypetcare/webservice/entity/EventEntity.java (hunk 10 above)
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/entity/EventEntity.java:42]: Codacy Issue found: [More than 5 parameters (found 9).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 14
           

Complexity increasing per file
==============================
- src/main/java/org/pesmypetcare/webservice/entity/EventEntity.java  1
- src/main/java/org/pesmypetcare/webservice/service/GoogleCalendarServiceImpl.java  1
- src/main/java/org/pesmypetcare/webservice/jsonhandler/DateTimeHandler.java  2
- src/main/java/org/pesmypetcare/webservice/error/InitializationException.java  1
- src/test/java/org/pesmypetcare/webservice/service/GoogleCalendarServiceTest.java  1
- src/main/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestController.java  1
- src/main/java/org/pesmypetcare/webservice/error/CalendarAccessException.java  1
- src/main/java/org/pesmypetcare/webservice/firebaseservice/CalendarServiceFactory.java  2
- src/test/java/org/pesmypetcare/webservice/dao/GoogleCalendarDaoTest.java  1
- src/test/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestControllerTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404&bid=17713034)

File: src/test/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestControllerTest.java (hunk 3 above)
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestControllerTest.java:53]: Codacy Issue found: [Method length is 22 lines (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/test/java/org/pesmypetcare/webservice/dao/GoogleCalendarDaoTest.java (hunk 14 above)
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/dao/GoogleCalendarDaoTest.java:69]: Codacy Issue found: [The String "calendarId" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/main/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestController.java [hunk 18]
```diff
@@ -0,0 +1,142 @@
...
+     * @param body Body of the request containing the id of the event to retrieve with key eventId assigned
+     * @return Event retrieved
+     * @throws CalendarAccessException If an error occurs when accessing the calendar
+     * @throws DatabaseAccessException If an error occurs when accessing the database
+     */
+    @GetMapping("/event/{owner}/{petName}")
+    public EventEntity retrieveEvent(@RequestHeader(TOKEN) String accessToken, @PathVariable String owner,
+                              @PathVariable String petName, @RequestBody Map<String, Object> body)
+        throws CalendarAccessException, DatabaseAccessException {
+        return googleCalendarService.retrieveEvent(accessToken, owner, petName, (String) body.get("eventId"));
```
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestController.java:107]: Codacy Issue found: [The String "eventId" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/test/java/org/pesmypetcare/webservice/service/GoogleCalendarServiceTest.java (hunk 11 above)
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/service/GoogleCalendarServiceTest.java:54]: Codacy Issue found: [NCSS for this method is 19 (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/main/java/org/pesmypetcare/webservice/entity/EventEntity.java [hunk 19]
```diff
@@ -0,0 +1,87 @@
...
+        this.location = location;
+        this.description = description;
+        this.color = color;
+        this.emailReminderMinutes = emailReminderMinutes;
+        this.repetitionInterval = repetitionInterval;
+        this.startDate = startDate;
+        this.endDate = endDate;
+    }
+
+    public Event convertToEvent() {
```
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/entity/EventEntity.java:56]: Codacy Issue found: [Method length is 21 lines (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/test/java/org/pesmypetcare/webservice/service/GoogleCalendarServiceTest.java (hunk 6 above)
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/service/GoogleCalendarServiceTest.java:72]: Codacy Issue found: [The String "Europe/Madrid" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/test/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestControllerTest.java (hunk 5 above)
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestControllerTest.java:61]: Codacy Issue found: [The String "{\n" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/test/java/org/pesmypetcare/webservice/service/GoogleCalendarServiceTest.java (hunk 7 above)
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/service/GoogleCalendarServiceTest.java:60]: Codacy Issue found: [The String "My eventId" appears 3 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/test/java/org/pesmypetcare/webservice/service/GoogleCalendarServiceTest.java (hunk 11 above)
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/service/GoogleCalendarServiceTest.java:54]: Codacy Issue found: [Method length is 24 lines (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/main/java/org/pesmypetcare/webservice/entity/EventEntity.java [hunk 20]
```diff
@@ -0,0 +1,87 @@
...
+        Event event = new Event()
+            .setId(id)
+            .setSummary(summary)
+            .setLocation(location)
+            .setDescription(description)
+            .setColorId(color);
+
+        EventDateTime start = new EventDateTime()
+            .setDateTime(startDate)
+            .setTimeZone("Europe/Madrid");
```
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/entity/EventEntity.java:66]: Codacy Issue found: [The String "Europe/Madrid" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/main/java/org/pesmypetcare/webservice/entity/EventEntity.java [hunk 21]
```diff
@@ -0,0 +1,87 @@
...
+        id = event.getId();
+        summary = event.getSummary();
+        location = event.getLocation();
+        description = event.getDescription();
+        color = event.getColorId();
+        startDate = event.getStart().getDateTime();
+        endDate = event.getEnd().getDateTime();
+    }
+
+    public EventEntity(String id, String summary, String description, String location, String color,
```
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/entity/EventEntity.java:42]: Codacy Issue found: [More than 5 parameters (found 9).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/test/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestControllerTest.java (hunk 1 above)
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestControllerTest.java:71]: Codacy Issue found: [The String "}" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/test/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestControllerTest.java (hunk 12 above)
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestControllerTest.java:80]: Codacy Issue found: [The String "/" appears 10 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

File: src/test/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestControllerTest.java (hunk 16 above)
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/controller/appmanager/GoogleCalendarRestControllerTest.java:104]: Codacy Issue found: [The String "/event/" appears 4 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/org/pesmypetcare/webservice/error/InitializationException.java  1
- src/main/java/org/pesmypetcare/webservice/firebaseservice/CalendarServiceFactory.java  2
- src/main/java/org/pesmypetcare/webservice/service/GoogleCalendarServiceImpl.java  1
- src/main/java/org/pesmypetcare/webservice/jsonhandler/DateTimeHandler.java  2
- src/main/java/org/pesmypetcare/webservice/error/CalendarAccessException.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404&bid=17713034)

**Review by AlvarTB — APPROVED** [review]
I don't see any mistake worth mentioning.

**Review by oricat8 — APPROVED** [review]
OK!

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage decreased per file
===========================
- src/main/java/org/pesmypetcare/webservice/error/RestExceptionHandler.java  -12
         

Complexity increasing per file
==============================
- src/main/java/org/pesmypetcare/webservice/jsonhandler/DateTimeHandler.java  2
- src/main/java/org/pesmypetcare/webservice/service/GoogleCalendarServiceImpl.java  1
- src/main/java/org/pesmypetcare/webservice/error/CalendarAccessException.java  1
- src/main/java/org/pesmypetcare/webservice/error/InitializationException.java  1
- src/main/java/org/pesmypetcare/webservice/firebaseservice/CalendarServiceFactory.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5403404&bid=17713034)

**Pull request merged**